patches/tModLoader/Terraria/ModLoader/ModContent.cs

```diff
@@ -490,6 +497,7 @@ internal static bool TryGetTexture(string name, out Asset<Texture2D> texture)
 			ModNet.Unload();
 			Config.ConfigManager.Unload();
 			CustomCurrencyManager.Initialize();
+			ItemRarity.Initialize();
```

Comment:
Calling vanilla methods in this class is weird. Make an Initialize() method in RarityLoader, and call ItemRarity.Initialize() in both rarityloader's initialize and unload. And remove both of its calls here.